x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/SettingsConfig.java

```diff
@@ -0,0 +1,186 @@
+            builder.field(TransformField.MAX_PAGE_SEARCH_SIZE.getPreferredName(), maxPageSearchSize);
+        }
+        if (docsPerSecond != null && (docsPerSecond.equals(DEFAULT_DOCS_PER_SECOND) == false)) {
+            builder.field(TransformField.DOCS_PER_SECOND.getPreferredName(), docsPerSecond);
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+        if (other == null || other.getClass() != getClass()) {
+            return false;
+        }
+
+        SettingsConfig that = (SettingsConfig) other;
+        return Objects.equals(maxPageSearchSize, that.maxPageSearchSize) && Objects.equals(docsPerSecond, that.docsPerSecond);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(maxPageSearchSize, docsPerSecond);
+    }
+
+    public static SettingsConfig fromXContent(final XContentParser parser, boolean lenient) throws IOException {
+        return lenient ? LENIENT_PARSER.apply(parser, null) : STRICT_PARSER.apply(parser, null);
+    }
+
+    public static class Builder {
+        private Integer maxPageSearchSize;
+        private Float docsPerSecond;
+
+        /**
+         * Default builder
+         */
+        public Builder() {
+
+        }
+
+        /**
+         * Builder starting from existing settings as base, for the purpose of partially updating settings.
+         *
+         * @param base base settings
+         */
+        public Builder(SettingsConfig base) {
+            this.maxPageSearchSize = base.maxPageSearchSize;
+            this.docsPerSecond = base.docsPerSecond;
+        }
+
+        /**
+         * Sets the paging maximum paging maxPageSearchSize that transform can use when
+         * pulling the data from the source index.
+         *
+         * If OOM is triggered, the paging maxPageSearchSize is dynamically reduced so that the transform can continue to gather data.
+         *
+         * @param maxPageSearchSize Integer value between 10 and 10_000
+         * @return the {@link Builder} with the paging maxPageSearchSize set.
+         */
+        public Builder setMaxPageSearchSize(Integer maxPageSearchSize) {
+            this.maxPageSearchSize = maxPageSearchSize;
+            return this;
+        }
+
+        /**
+         * Sets the docs per second that transform can use when pulling the data from the source index.
+         *
+         * This setting throttles transform by issuing queries less often, however processing still happens in
+         * batches. A value of 0 disables throttling (default).
+         *
+         * @param docsPerSecond Integer value
+         * @return the {@link Builder} with requestsPerSecond set.
+         */
+        public Builder setRequestsPerSecond(Float docsPerSecond) {
+            this.docsPerSecond = docsPerSecond;
```

Comment:
It seems to me that a `null` passed in here is where you want to set the default values back. Otherwise, it becomes complicated to know when a `null` means "not supplied" vs "revert to default"

Similar comment for `setMaxPageSearchSize`

Reply:
I just found this to be a problem in hlrc (and fixed it there, not pushed yet).

However I wonder if this is really a problem in this place, both setters are not used, but only update. Maybe I rename this class to Updater and remove the setters.